crates/ra_mbe/src/lib.rs

```diff
@@ -36,11 +45,11 @@ pub struct MacroRules {
 }
 
 impl MacroRules {
-    pub fn parse(tt: &tt::Subtree) -> Option<MacroRules> {
+    pub fn parse(tt: &tt::Subtree) -> Result<MacroRules> {
         mbe_parser::parse(tt)
     }
-    pub fn expand(&self, tt: &tt::Subtree) -> Option<tt::Subtree> {
-        mbe_expander::exapnd(self, tt)
+    pub fn expand(&self, tt: &tt::Subtree) -> Result<tt::Subtree> {
+        mbe_expander::expand(self, tt)
     }
```

Comment:
I think it makes sense to use different error type during parsing and during expansion, because the possible error conditions are different. SO, we need `enum MacroRulesParsingError` and `MacroRulesExpansionError`. Given that we are in the `mbe` crate, I think `ParseError` and `ExpandError` would be better names.